src/vm/eventpipe.cpp

```diff
@@ -511,12 +517,22 @@ void EventPipe::WriteEventInternal(EventPipeEvent &event, EventPipeEventPayload
 
             if(s_pFile != NULL)
             {
-                s_pFile->WriteEvent(instance);
+                // EventPipeFile::WriteEvent needs to allocate a metadata event
+                // and can therefore throw. In this context we will silently
+                // fail rather than disrupt the caller
+                EX_TRY
+                {
```

Comment:
If WriteEvent needs to allocate, it should probably nothrow allocate and check the value.  Then, we don't have to try/catch, and it declares the intent more clearly.

That said, I remember you telling me you were having trouble getting nothrow new to work in some places.  Is this one of those?

Reply:
There are a number of places below Disable which allocate. They could all be transferred to use (nothrow) with some effort, but my concern becomes: Where and how do we report errors? In shutdown we certainly don't want to report an error by throwing, but if a user calls Disable and it fails (for instance, in rundown) how do we report that to the user? If we want to use nothrow everywhere then do we start using HResults to report a called function call due to an allocation at some depth?

Reply:
It's a good point.  I'm not sure how we want to handle this.  Perhaps that deserves its own issue, since you'll still get the same nothrow behavior from the code change here.  So, really, we have to decide if we want to throw or not.  For this, I'd say go with what you have, and file an issue to follow-up on what the behavior of disable should be.

Reply:
I got mixed up when I wrote the comment above I now realized. The real reason why I have a catch here is not for the allocations further down, but because SString::GetUnicode and SHash::Add both throw. This could be moved down to cover those spots, with a decent amount of extra code to handle the edge cases when those operations fail, but I did not do so at first. Would it be better to do it that way?

Reply:
Also this code has nothing to do with EventPipe::Disable

Reply:
I think it's OK as is.